### PR TITLE
Bump `chrono`, fix deprecation warnings and a clippy issue

### PR DESCRIPTION
**Pull request by suryapandian**
## Motivation

Trivial crate updates and clippy fixes
Update chrono function following their [deprecation warning](https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDateTime.html#method.from_timestamp)

### CODE REVIEW

File: examples/pod_shell_crossterm.rs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 use futures::{channel::mpsc::Sender, SinkExt, StreamExt};
 use k8s_openapi::api::core::v1::Pod;
 
-#[cfg(unix)] use crossterm::event::Event;
```
Comment (clux) [inline examples/pod_shell_crossterm.rs:4]: what's up with this? this will likely break windows

File: kube-derive/tests/crd_schema_test.rs [hunk 2]
```diff
@@ -132,7 +132,7 @@ fn test_serialized_matches_expected() {
             nullable: None,
             nullable_skipped_with_default: None,
             nullable_with_default: None,
-            timestamp: DateTime::from_utc(NaiveDateTime::from_timestamp(0, 0), Utc),
+            timestamp: DateTime::from_utc(NaiveDateTime::from_timestamp_opt(0, 0).unwrap(), Utc),
```
Comment (clux) [inline kube-derive/tests/crd_schema_test.rs:135]: this looks sensible given the [deprecation warning](https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDateTime.html#method.from_timestamp)

**Review by clux — COMMENTED** [review]
think the crossterm removal is probably an accident?

File: examples/pod_shell_crossterm.rs (hunk 1 above)
Comment (suryapandian) [reply examples/pod_shell_crossterm.rs:4]: when i build I got the warning
```
warning: unused import: `crossterm::event::Event`
 --> examples/pod_shell_crossterm.rs:4:18
  |
4 | #[cfg(unix)] use crossterm::event::Event;
  |                  ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
  ```
  
  maybe we should use `#[warn(unused_imports)]` instead if needed, I will update it likewise if the CI breaks 🤔

Comment (clux) [reply examples/pod_shell_crossterm.rs:4]: hm, actually, this actually is unused. i thought it was some weirdness with an imported trait vs mac use or something. But this is not needed as it's just an unused enum, and terminal resize works without it. Probably an oversight in clippy not correctly running on forks.

**Review by clux — APPROVED** [review]

**Review by clux — APPROVED** [review]

**Comment by clux** [comment]
Thanks!

**Pull request merged**